usecases/traverser/near_params_vector.go

```diff
@@ -0,0 +1,195 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2021 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package traverser
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/pkg/errors"
+	"github.com/semi-technologies/weaviate/entities/additional"
+	"github.com/semi-technologies/weaviate/entities/modulecapabilities"
+	"github.com/semi-technologies/weaviate/entities/near"
+	"github.com/semi-technologies/weaviate/entities/schema/crossref"
+	"github.com/semi-technologies/weaviate/entities/search"
+)
+
+type nearParamsVector struct {
+	modulesProvider ModulesProvider
+	search          nearParamsSearcher
+}
+
+type nearParamsSearcher interface {
+	ObjectByID(ctx context.Context, id strfmt.UUID,
+		props search.SelectProperties, additional additional.Properties) (*search.Result, error)
+}
+
+func newNewParamsVector(modulesProvider ModulesProvider, search nearParamsSearcher) *nearParamsVector {
+	return &nearParamsVector{modulesProvider, search}
+}
+
+func (e *nearParamsVector) vectorFromParams(ctx context.Context,
```

Comment:
why `e` in `(e *nearParamsVector)`?

Reply:
not sure why this was the case. changed it to `v`